### ISSUE TITLE
Weird UP button key behavior with Resident Evil 0

### ISSUE THREAD

**Issue by anjari** [KrossX/Durazno#36]
I'm using xbox360 wired controller on Windows 10.

First, I tried x360ce for RE0 but the key remapping doesn't work at all. It has no effect at all. The game runs with normal key mapping.

Then I tried Durazno (latest version), I followed the instruction and copy the DLL and INI file to the game EXE directory. I only swapped the X and B key. After run the game, I found weird behavior on the controller key. When I pressed UP button on menu screen, it will perform A button. It's very strange as I didn't remap UP button at all.

On the game, when I pressed UP button, Rebecca will do aim to up and shoot, then pause menu is invoked at almost same time. I am thinking the UP button was remapped into all buttons somehow.

I reverted the key mapping to use default key mapping, the problem with UP button still occurs.

**Comment by KrossX** [comment]
I can't reproduce it. Could you post your ini file? Also, what happens when you use the test on the Durazno GUI?

**Comment by anjari** [comment]
Ok, somehow after I installed "DirectX End-User Runtime (June 2010)" (which is one of requirement of x360ce), the issue disappeared (x360ce issue is still not solved though).

Is directx runtime required for the Durazno?

**Comment by KrossX** [comment]
It requires XInput support, same as the game.  There are some files included with the OS and others are included with DirectX ([more details here](https://msdn.microsoft.com/en-us/library/windows/desktop/hh405051(v=vs.85).aspx)). You should've already have it though since DirectX runtimes are usually installed with the game or in Steam's case, on first run.

**Comment by anjari** [comment]
Hi, the problem just appeared again. It's had been two weeks since the last time I played the game. Today I just started the game and found out the issue apeeared again. Any idea how to troubleshoot the issue? Is there logging on the dll ?

**Comment by KrossX** [comment]
So it's only a problem with RE0 and only if you use remapping? When you tested Alien Isolation, did you copy the same ini file used with RE0? Can you attach your ini file to see if there's anything weird there?

**Comment by anjari** [comment]
Yes, I use same ini file. I only swap the X and B button. Here is the INI file contents:

```
[General]
INIversion=4
LoadDLL=
[Controller0]
Port=0
Disable=0
Dummy=0
Rumble=1000
L_LinearDZ=0
L_LinearADZ=0
L_Inverted_X=0
L_Inverted_Y=0
L_Linearity=300
L_Deadzone=0
L_AntiDeadzone=0
R_LinearDZ=0
R_LinearADZ=0
R_Inverted_X=0
R_Inverted_Y=0
R_Linearity=300
R_Deadzone=0
R_AntiDeadzone=0
L_TriggerMin=0
L_TriggerMax=255
R_TriggerMin=0
R_TriggerMax=255
Remap=00 01 02 03 04 05 06 07 08 09 10 12 11 13 14 15 16 17 18 19 20 21 22 23
[Controller1]
Port=1
Disable=0
Dummy=0
Rumble=1000
L_LinearDZ=0
L_LinearADZ=0
L_Inverted_X=0
L_Inverted_Y=0
L_Linearity=300
L_Deadzone=0
L_AntiDeadzone=0
R_LinearDZ=0
R_LinearADZ=0
R_Inverted_X=0
R_Inverted_Y=0
R_Linearity=300
R_Deadzone=0
R_AntiDeadzone=0
L_TriggerMin=0
L_TriggerMax=255
R_TriggerMin=0
R_TriggerMax=255
Remap=00 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17 18 19 20 21 22 23
[Controller2]
Port=2
Disable=0
Dummy=0
Rumble=1000
L_LinearDZ=0
L_LinearADZ=0
L_Inverted_X=0
L_Inverted_Y=0
L_Linearity=300
L_Deadzone=0
L_AntiDeadzone=0
R_LinearDZ=0
R_LinearADZ=0
R_Inverted_X=0
R_Inverted_Y=0
R_Linearity=300
R_Deadzone=0
R_AntiDeadzone=0
L_TriggerMin=0
L_TriggerMax=255
R_TriggerMin=0
R_TriggerMax=255
Remap=00 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17 18 19 20 21 22 23
[Controller3]
Port=3
Disable=0
Dummy=0
Rumble=1000
L_LinearDZ=0
L_LinearADZ=0
L_Inverted_X=0
L_Inverted_Y=0
L_Linearity=300
L_Deadzone=0
L_AntiDeadzone=0
R_LinearDZ=0
R_LinearADZ=0
R_Inverted_X=0
R_Inverted_Y=0
R_Linearity=300
R_Deadzone=0
R_AntiDeadzone=0
L_TriggerMin=0
L_TriggerMax=255
R_TriggerMin=0
R_TriggerMax=255
Remap=00 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17 18 19 20 21 22 23
```

**Comment by KrossX** [comment]
So it's now a general issue with RE0 then. I was hopping it was a bug with remapping. Which DLL are you using? Only xinput1_3 should be used for the game I think.

**Comment by anjari** [comment]
I copied all of them. I did try using only xinput1_3.dll. Same result.

**Comment by KrossX** [comment]
Thanks for trying, that rules out some possible problems I thought. I'll see if I can check the game myself later today and post my results. With luck, it's reproducible.

**Comment by KrossX** [comment]
I could not reproduce it. I used your INI file, the buttons were swapped and everything was fine. =(

In the gamepad status data "dpad-up" is the very first bit, to have that problem something must be writing memory it shouldn't or some zero memory ain't zero. Here's a dll that applies no transformation, and a dll that has the "dummy" option disabled. I'm just guessing here so good luck.

[TEST1-NoTransform](https://drive.google.com/open?id=0B1wssOeJs5_7WFF0TWRhdVhIWDQ), [TEST2-NoDummy](https://drive.google.com/open?id=0B1wssOeJs5_7clQtVFp0cF9CR1k)

PS: Both dlls are compiled with a different and very old compiler, so here's a "default" build just in case. This DLL should have just as bad as the one you've been testing with. [TEST3-Control](https://drive.google.com/open?id=0B1wssOeJs5_7MTBYcjd3UlVCS3c)

**Comment by anjari** [comment]
Hi, good news, two of your Test DLL are working: TEST2-NoDummy and TEST3-Control. For TEST1-NoTransform, the key remap doesn't work (it loads default keys). All of them don't have UP key issue for sure. With this result, could you figure out what would cause the issue?

**Comment by anjari** [comment]
So far using the TEST3 DLL there is no issue.

**Comment by KrossX** [comment]
Thanks, it was a compilation problem indeed then. I still don't have a proper setup, that dll was compiled with the trusty old 98's compiler which sadly cannot make 64bit dlls.
